### ISSUE TITLE
Omnimount Container Error: "Connection Refused"

### ISSUE THREAD

**Issue by Misterobots** [kelinger/OmniStream#49]
I have no idea what could be happening here. I've changed nothing and everything was working a day ago. Any insight or assistance is appreciated.
![image](https://user-images.githubusercontent.com/9805557/225144923-cfe07450-7adb-4a73-ae6f-98595727918a.png)

**Comment by elmerfreak** [comment]
Same here, happen since yesterday, no reason

**Comment by kelinger** [comment]
First question:  have either of you tried a full stop/start of OmniStream?

```
omni down
omni up
```

**Comment by elmerfreak** [comment]
Yes, and still the problem.

I was looking in this issue from [rclone](https://forum.rclone.org/t/unable-to-upload-to-gsuite-as-of-a-couple-hours-ago/35755), change my DNS but still no work.


<img width="348" alt="Screenshot 2023-03-14 at 23 17 03" src="https://user-images.githubusercontent.com/5510709/225213738-6a9a1a8e-063d-4851-881d-2ff0276e4fb1.png">

<img width="742" alt="Screenshot 2023-03-14 at 23 17 34" src="https://user-images.githubusercontent.com/5510709/225213792-548f8caa-586f-4bb5-8f9b-3fe6cbe255e5.png">

**Comment by Misterobots** [comment]
> Well, that may be a problem. It looks like it is using fusermount3 which is apparently a major upgrade (ie, new commands/options/parameters).
> 
> On your host, what happens when you type `fusermount -V` (that's a captial V)?


"fusermount version: 2.9.9"


What caused this? Is an update to the Omnimount container? 

Thank you for the assistance.

**Comment by elmerfreak** [comment]
> fusermount -V

fusermount version: 2.9.9

**Comment by Misterobots** [comment]
It does look like Rclone moved to Fuse3 in the last major update(v1.62.0). Does this coincide with the behavior?

**Comment by kelinger** [comment]
Definitely an issue with Rclone.  See:  https://github.com/rclone/rclone/issues/6844

OmniMount has a version of rclone baked into it.  However, since I don't update the OmniMount container that frequently (there's not much need to) it does check for newer versions of Rclone when it is started up.  It isn't looking down the beta path so the assumption was that things making it out of beta and into a brand new release have been pretty well vetted.  Obviously, this one snuck through!

The reason it doesn't affect everyone is that, I'm guessing, you both performed a restart of at least OmniMount roughly the same day while I have left mine up and running for a while now.

It looks like he pushed a fix out today.  I'm tempted to take down my (otherwise working) system and see if the newest release of rclone fixes it.  If not, I can build a new OmniMount that either doesn't upgrade or allows you to override the upgrade.  I'd rather not eliminate it, though, because this is how I can insure that things stay reasonably functional and updated in case I can't create a new build immediately after new releases.

**Comment by elmerfreak** [comment]
Thanks @kelinger for your help.

I installed already the rclone beta version pushed today but still no working, also downgrade rclone and no luck, install fuse3 no luck again. :(

**Comment by kelinger** [comment]
Right; the problem is that you probably installed it on your host, but my container uses the version installed there and that's not as easy for you to update.

I just took down my system and it's staying down so... yeah... this is personal now.  I should have a fixed version of OmniMount in the next few hours.

**Comment by elmerfreak** [comment]
Oh God!! at least we are in the same boat now 🤣

**Comment by kelinger** [comment]
Fix:  `omni update`

OmniMount has been recompiled to use Fuse3 and the latest Rclone build.  The beta with the patch wasn't released yet (as of this writing) so I just updated the container to use Fuse3 and it tested successfully.

To verify you're using the latest OmniMount:  `docker exec -ti omnimount fusermount -V`
The response should be something like:  fusermount3 version: 3.10.3

If the "omni update" command doesn't work for you, you may try `omni clean` followed by `omni up`.

**Comment by Misterobots** [comment]
Confirmed working and everything back up on my server. Thank you @kelinger  for the quick attention and speedy resolution!

**Comment by elmerfreak** [comment]
It work for me too! thanks @kelinger 


Now I'm still having ban from Google apis 👎  any idea how deal with it? I don't know if open other issue 😞 
> I was looking in this issue from [rclone](https://forum.rclone.org/t/unable-to-upload-to-gsuite-as-of-a-couple-hours-ago/35755), change my DNS but still no work.

**Comment by TechPerplexed** [comment]
I used to run into a lot of bans too, especially when I scanned a lot of content at once. I think the only solution is to ride it out - it'll probably be lifted in a few hours.

**Comment by kelinger** [comment]
If you run `tail $LOGS/rclone.log` you should see the last ~10 rclone errors.  Note that this might be filled with the "fusermount" errors mentioned in this issue and if that's the last one, then you're probably not banned anymore (or not running anything).

But, if you see file operations and something like "permission denied" or "unauthorized" it likely means you're still banned.  The longest I've been banned has been 24 hours (this assumes I stopped doing whatever caused it).

You could also run the command `api` which should show the amount of data that rclone has copied to Google in the past 24 hours (if you are accessing Google with the same user ID elsewhere, that is not included in this tabulation).  If it's high, this can occur when someone refreshes a movie collection or long TV series.  I've had it happen when someone reposted the complete James Bond collection and another time when they refreshed a TV series with over 200 episodes across 10+ seasons.

**Comment by christeasdale** [comment]
I picked a rough time to finally upgrade from gooby.  I have most everything up and working on OmniStream except i cannot get the default Media folder to change location.  I have my Google drive folder structure set for use with sever different applications so I don't want to change anything there.  If I setup /Media and throw a couple temp media files in there it works fine.  The folder where my media also shows up in the cloud folder inside mnt folder.  When I try "Redefine Media Folder" in Omni settings it just goes black for a second and returns to the menu.  No option to select a folder from my GD.  My account is configured with the media on a shared TeamDrive.  Is there a manual way to do this that won't get reset?  Thanks

**Comment by elmerfreak** [comment]
The message is: 
`We're sorry... but your computer or network may be sending automated queries. To protect our users, we can't process your request right now.` 

And its weird, the api command show something crazy
`Since 2023-03-15 05:29:10
Gigabytes:  49410.073131115176
Files:      23673`

Its imposible transfer 49T in one day with Google 🤔

**Comment by elmerfreak** [comment]
Done, hope it help

<img width="582" alt="Screenshot 2023-03-16 at 19 08 27" src="https://user-images.githubusercontent.com/5510709/225786543-d81b4880-e360-41a2-aec6-829fd34bbd01.png">

**Comment by thunderstorm654** [comment]
> The message is: `We're sorry... but your computer or network may be sending automated queries. To protect our users, we can't process your request right now.`
> 
> And its weird, the api command show something crazy `Since 2023-03-15 05:29:10 Gigabytes: 49410.073131115176 Files: 23673`
> 
> Its imposible transfer 49T in one day with Google 🤔

I have the same issue with OmniStream and rclone right now when I checked the rclone logs. Nothing will play at the moment and the logs are full of these automated queries errors I've never seen before. Did shutting down your mount for a while sort it out?

Confirmed I am running latest version fusermount3 3.10.3.

Thanks!

**Comment by elmerfreak** [comment]
Well, I disabled the cron for about 48h, enabled again but still the same, poorly it upload the backup files 😞 

Maybe is the end of GDrive as it is. 

Hope @kelinger has new ideas and came to rescue 😃

**Comment by thunderstorm654** [comment]
> Well, I disabled the cron for about 48h, enabled again but still the same, poorly it upload the backup files 😞
> 
> Maybe is the end of GDrive as it is.
> 
> Hope @kelinger has new ideas and came to rescue 😃

What hosting provider are you using? Is it Hetzner by any chance? I found that a lot of people are getting this automated queries error on their servers on and off the past few months, particularly bad the past week or so. Apparently it looks like Google may be throttling/banning Hetzner IP ranges as there is excessive traffic from them. 

People have tried some workarounds like changing DNS or adding things to hosts file but these don't always work for long. Mine is working again now for about a day, but I'm not convinced it'll just break again at any point. Might move provider to be honest. Pain!!! I hope it doesn't spread to other providers and gradually Google drive is unusable. I have a lot of content..!!

**Comment by elmerfreak** [comment]
Yeah, its Hetzner, Im trying with other provider and works great, as you said hope Google don't start a witch hunt banning every provider.

> Currently all I did is add the entries to etc/hosts mentioned in this reply:
> https://forum.rclone.org/t/429-google-drive-errors-back-for-anyone-using-hetzner-even-with-cloudflare-or-google-dns-used/36671/62?u=storm

I did this as well but no work.

Plex works as the cron is disabled and no start evary 15 min.

**Comment by kelinger** [comment]
@elmerfreak did you ever get the Hetzner issue resolved?

**Comment by elmerfreak** [comment]
Hi @kelinger not really, Google block completely my IP, I did everything posting on rclone and reddit forums, the only way was making wireguard and tunnelin the google traffic, so I myself decide split my providers, one for Plex other for download flow, but It became mor expence.

So, think on that, OmniStream work smoth with other storage companies like Dropbox or Mega for example?

**Comment by kelinger** [comment]
@elmerfreak - Did this resolve for you with Google?  I haven't experienced (or read about others experiencing) that with other providers.
